### PR TITLE
cherry-pick from master to dev-1.x

### PR DESCRIPTION
**Pull request by lvhan028**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Modification

- [x] #1275 
- [x] #1238 
- [x] #1261 
- [x] #1308 
- [x] #1314
- [x] #1343  

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: mmdeploy/backend/coreml/torchscript2coreml.py [hunk 1]
```diff
@@ -47,30 +50,22 @@ def create_shape(name: str, input_shapes: Dict) -> ct.Shape:
...
                                               torch.jit.RecursiveScriptModule],
-                     output_file_prefix: str,
-                     input_names: Sequence[str],
-                     output_names: Sequence[str],
-                     input_shapes: Dict,
-                     convert_to: str = 'neuralnetwork',
-                     fp16_mode: bool = False,
-                     skip_model_load: bool = True,
-                     **kwargs):
+                     output_file_prefix: str, deploy_cfg: Union[str,
```
Comment (grimoire) [inline mmdeploy/backend/coreml/torchscript2coreml.py:56]: Did we decide NOT accept deploy_cfg as the input?

File: mmdeploy/backend/coreml/torchscript2coreml.py [hunk 2]
```diff
@@ -85,32 +80,52 @@ def from_torchscript(torchscript_model: Union[str,
...
         compute_precision=compute_precision,
         convert_to=convert_to,
-        skip_model_load=False)
+        minimum_deployment_target=ct.target[minimum_deployment_target]
+        if minimum_deployment_target else None,
+        skip_model_load=final_params.get('skip_model_load', False))
 
     suffix = get_model_suffix(convert_to)
     output_path = output_file_prefix + suffix
+    backend_files.append(output_path)
```
Comment (grimoire) [inline mmdeploy/backend/coreml/torchscript2coreml.py:130]: Not a good idea to update backend_files here

File: mmdeploy/backend/coreml/torchscript2coreml.py (hunk 1 above)
Comment (lvhan028) [reply mmdeploy/backend/coreml/torchscript2coreml.py]: My mistake. PR #1308 will not be merged to dev-1.x until it is fixed in master branch

**Review by grimoire — APPROVED** [review]

**Review by lzhangzz — APPROVED** [review]
LGTM

**Review by irexyc — APPROVED** [review]

**Review by AllentDan — COMMENTED** [review]
LGTM

**Pull request merged**